### PR TITLE
Publish by channel_id

### PR DESCRIPTION
**Pull request by daovist**
Issue #1433 

Add `channelId` to `PublishState` and use it to replace `channel` in `PublishParams`. Store both in redux while editing but always use `channel_id` to publish.

Adds `myChannels: selectMyChannelClaims(state)` to PublishForm and uses it to get a channel's id from its name when saving form data.

@tzarebczan check this out. I am able to create a new channel and publish seconds later

lbry fast publish

### CODE REVIEW

**Comment by tzarebczan** [comment]
@daovist - nice work! https://github.com/lbryio/lbry-redux/pull/26 should fix the channel selection issue.  I'll test this out once that's merged into redux

File: src/renderer/component/publishForm/view.jsx [hunk 1]
```diff
@@ -137,13 +139,16 @@ class PublishForm extends React.PureComponent<Props> {
   }
 
   handleChannelChange(channelName: string) {
-    const { name, updatePublishForm } = this.props;
+    const { name, updatePublishForm, myChannels } = this.props;
+    const form = { channel: channelName };
+    const namedChannelClaim = myChannels.filter(channel => channel.name === channelName)[0];
+    if (namedChannelClaim) {
+      form.channelId = namedChannelClaim.claim_id;
```
Comment (seanyesmunt) [inline src/renderer/component/publishForm/view.jsx]: This should clear out the channel id if there is no `namedChannelClaim`

If I select a channel, then go back to anonymous, the same `claim_id` will be stored in redux

**Comment by seanyesmunt** [comment]
Looks like this was already a bug. I thought I fixed it but I guess not.

**Comment by seanyesmunt** [comment]
Looks like `channel` isn't being correctly used to create `uri` in the publish reducer.

File: src/renderer/component/publishForm/view.jsx [hunk 2]
```diff
@@ -137,13 +139,14 @@ class PublishForm extends React.PureComponent<Props> {
   }
 
   handleChannelChange(channelName: string) {
-    const { name, updatePublishForm } = this.props;
+    const { name, updatePublishForm, myChannels } = this.props;
+    const form = { channel: channelName };
+    const namedChannelClaim = myChannels.filter(channel => channel.name === channelName)[0];
```
Comment (skhameneh) [inline src/renderer/component/publishForm/view.jsx]: use `Array.prototype.find()`? (instead of filter)

Comment (daovist) [reply src/renderer/component/publishForm/view.jsx]: Good call @skhameneh

**Pull request merged**